app/src/main/java/com/preonboarding/sensordashboard/common/di/RepositoryModule.kt

```diff
@@ -0,0 +1,25 @@
+package com.preonboarding.sensordashboard.common.di
+
+import com.preonboarding.sensordashboard.data.local.datasource.SensorHistoryLocalDataSource
+import com.preonboarding.sensordashboard.data.repositoryimpl.SensorHistoryRepositoryImpl
+import com.preonboarding.sensordashboard.domain.repository.SensorHistoryRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+
+@Module
+@InstallIn(SingletonComponent::class)
+object RepositoryModule {
+
+    @Provides
+    @Singleton
+    fun provideSensorHistoryRepository(
+        sensorHistoryLocalDataSource: SensorHistoryLocalDataSource,
+    ): SensorHistoryRepository {
+        return SensorHistoryRepositoryImpl(sensorHistoryLocalDataSource)
+    }
```

Comment:
object가 아닌 abstract class로 
@Provides 말구 @Binds를 하여 구현체와 인터페이스를 bind 해주는게 어떨까요?
```suggestion
@Module
@InstallIn(SingletonComponent::class)
abstract class RepositoryModule {

    @Binds
    @Singleton
    abstract fun bindSensorHistoryRepository(
        repositryImpl : SensorHistoryRepositoryImpl
    ): SensorHistoryRepository
```

Reply:
오오 혹시 이것에 대한 이점도 알려주실 수 있나요